src/AttendanceHandler.cpp

```diff
@@ -0,0 +1,69 @@
+
+#include <algorithm>
+#include "AttendanceHandler.hpp"
+
+auto AttendanceHandler::matchByEmail(const User& userProfile)
+{
+    return [&userProfile](const auto& report)
+    {
+        return !report.second.getEmail().compare(userProfile.getEmail());
+    };
+}
+
+void AttendanceHandler::reportAbsence(const User& userProfile,
+                                      const Lesson& Lesson,
+                                      const std::string& comment)
+{
+    report(userProfile, Lesson, "Absence", comment);
+}
+
+void AttendanceHandler::reportMakeUp(const User& userProfile,
+                                     const Lesson& Lesson,
+                                     const std::string& absenceDate,
+                                     const std::string& comment)
+{
+    report(userProfile, Lesson, "MakeUp for: " + absenceDate, comment);
+}
+
+AttendanceHandler::AttendanceList AttendanceHandler::viewReports(const User * const userProfile)
```

Comment:
I think that default agrument = nullptr would be useful

Reply:
nullptr is not a default value. 

I expected to receive a pointer to User object from anyone who is not a trainer, that way I do not consider nullptr to be a default value.